### ISSUE TITLE
[Bitnami Wordpress] Instance runs out of RAM and triggers oom-kill

### ISSUE THREAD

**Issue by cece99** [bitnami/vms#126]
### Platform

Google Cloud Platform

### bndiagnostic ID [know more about bndiagnostic ID](https://docs.bitnami.com/general/how-to/understand-bndiagnostic/)

271db450-2819-feb2-da2a-fa46a910439c

### bndiagnostic output

    ? Wordpress: Found possible issues
    ? Apache: Found possible issues
    ✓ Processes: No issues found
    ✓ Connectivity: No issues found
    ? Mariadb: Found possible issues
    ✓ Php: No issues found
    ? Resources: Found possible issues
 

 
[Wordpress]

Some files don't have the expected permissions Please check the following guide:
 
https://docs.bitnami.com/general/apps/wordpress/administration/understand-
file-permissions/

[Apache]

Press [Enter] to continue:
Found recent error or warning messages in the Apache error log.
```
[Fri Jul 15 14:45:41.266164 2022] [proxy_fcgi:error] [pid 1441:tid 
139774492874496] [client **ip_address**:61408] AH01079: failed to make 
connection to backend: httpd-UDS, referer: 
https://vacustore.it/wp-admin/tools.php?page=regenerate-thumbnails
 [Fri Jul 15 14:45:41.371176 2022] [proxy:error] [pid 1441:tid 139774459303680] 
(2)No such file or directory: AH02454: FCGI: attempt to connect to Unix domain 
socket /opt/bitnami/php/var/run/www.sock (www-fpm:8000) failed
 [Fri Jul 15 14:45:41.371220 2022] [proxy_fcgi:error] [pid 1441:tid 
139774459303680] [client **ip_address**:61410] AH01079: failed to make 
connection to backend: httpd-UDS, referer: 
https://vacustore.it/wp-admin/tools.php?page=regenerate-thumbnails

```
Please check the following guide to troubleshoot server issues:
 
https://docs.bitnami.com/general/apps/wordpress/troubleshooting/debug-erro
rs-apache/

A high number of incoming requests originate from one or more unique IP 
addresses. This could indicate a bot attack. The following guide shows how to 
check for and block suspicious IP addresses.
Press [Enter] to continue:
 
https://docs.bitnami.com/bch/apps/moodle/troubleshooting/deny-connections-
bots-apache/

[Mariadb]

Found recent error messages in the MariaDB error log:
```
2022-07-15 14:45:40 164 [Warning] Aborted connection 164 to db: 
'bitnami_wordpress' user: 'bn_wordpress' host: 'localhost' (Got an error reading 
communication packets)

```
Please check the following guide to troubleshoot MariaDB issues:
 
https://docs.bitnami.com/aws/apps/wordpress/troubleshooting/debug-errors-m
ariadb/

[Resources]

Your instance has little available RAM memory.
```
               total        used        free      shared  buff/cache   available
Press [Enter] to continue:
Mem:            3933        2837         232         107         863         742
Swap:              0           0           0
```
You could try to increase your instance's memory. Please check your cloud 
provider's documentation for more information.

You can also enable swap memory to improve performance.
 
https://docs.bitnami.com/installer/faq/linux-faq/administration/increase-m
emory-linux/
 

===== End of bndiagnostic tool output =====

### bndiagnostic was not useful. Could you please tell us why?

The instance has 4GB of RAM and does not have lots of traffic (50 visits per day) but still crashes.

### Describe your issue as much as you can

The instance has 2vCPUs and 4GB of RAM but sometimes Wordpress becomes irresponsive because bitnami.service is killed by oom-kill. It can happen after an hour or after a couple of days, but a way to trigger it is to run regenerate thumbnails and it crashed after regenerating around 50 thumbnails.
A very similar website (same theme, same plugins,..) is running on another instance (2vCPUs, 4GB of RAM) with bitnami Wordpress (an older version, from around 2 years ago) and never crashed.

**Comment by mdhont** [comment]
I recommend asking in the WordPress community for any known errors with regenerating thumbnails. It doesn't seem related to the Bitnami configuration.

**Comment by cece99** [comment]
regenerate thumbnail is an example and a way to trigger immediately the crash. If I don't use regenerate thumbnails after a 8-9 hours I find the website crashed because of oom-kill.

**Comment by mdhont** [comment]
Please contact the plugin support.

**Comment by NachoClapps** [comment]
@mdhont Would it be possible to update the GD Library manually on an Debian instance?

I've copied my Wordpress site which runs Bitnamy on AWS Lightsail to *2 different hostings*, one provided by Wetopi (Spain) and a Blogvault one.

I did not experience any memory issues performing the same exact tasks, with the same plug-in configuration, etc.

The only difference I could find was that they came *with GD Library 2.3* compared to Bitnami's GD Library, which reads: 

``` *bundled (2.1.0 compatible)*. ```

Perhaps this newer version is different and more optimized? I don't know.

Their PHP-FPM children-related settings seem to be more conservative as well.
